packages/util-endpoints/src/lib/parseURL.ts

```diff
@@ -8,21 +8,35 @@ const DEFAULT_PORTS: Record<EndpointURLScheme, number> = {
 };
 
 /**
- * Parses a string into it’s Endpoint URL components.
+ * Parses a string, URL, or Endpoint into it’s Endpoint URL components.
  */
-export const parseURL = (value: string): EndpointURL | null => {
+export const parseURL = (value: string | URL | Endpoint): EndpointURL | null => {
   const whatwgURL = (() => {
     try {
+      if (value instanceof URL) {
+        return value;
+      }
+      if (typeof value === "object" && "hostname" in value) {
+        const { hostname, port = "", protocol = "", path = "", query = {} } = value as Endpoint;
+        const url = new URL(`${protocol}//${hostname}${port ? `:${port}` : ""}${path}`);
+        url.search = Object.entries(query)
+          .map(([k, v]) => `${k}=${v}`)
+          .join("&");
+        return url;
+      }
       return new URL(value);
     } catch (error) {
       return null;
     }
   })();
 
   if (!whatwgURL) {
+    console.error(`Unable to parse ${JSON.stringify(value)} as a whatwg URL.`);
```

Comment:
I added a message here because the endpoints resolved error gives too little info.